### DISCUSSION TITLE
Change the scheduler interval in CLI

### DISCUSSION THREAD

**Discussion by ForJadeForest** [Lightning-AI/pytorch-lightning#13975] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I just trying to use `StepLR`  in my `.yaml` config. 
```yaml
lr_scheduler:
  class_path: StepLR
  init_args:
    step_size : 10
    gamma : 0.5
```
I guess the CLI will set the `interval` as "epoch", but I want to use the "step" to control my learning rate.  I do not find anything about it in the docs. How can achieve it just in my `.yaml` file?

**Comment by rohitgr7** [comment]
cc: @carmocca

**Comment by carmocca** [comment, marked as answer]
This is not possible from the yaml file. If you need to customize the learning rate scheduler configuration, you can do so by overriding:


```python
class MyLightningCLI(LightningCLI):
    @staticmethod
    def configure_optimizers(lightning_module, optimizer, lr_scheduler=None):
        if lr_scheduler is None:
            return optimizer
        return {
            "optimizer": optimizer,
            "lr_scheduler": {"scheduler": lr_scheduler, "interval": "step"},
        }
```

**Reply by ForJadeForest** [reply]
Thanks for your reply. Do you mean that I can change the interval in the `configure_optimizers` method? But if using it, can I change the optimizer from the yaml file?

**Reply by carmocca** [reply]
Yes, but notice this is the `configure_optimizers` method in the `LightningCLI` that will replace the method in the `LightningModule`. You can still change the optimizer instance itself from the config file. It will be passed as the `optimizer` argument in the snippet above.

**Reply by ForJadeForest** [reply]
That's what I need. Thanks a lot!
